src/Utils/ChatDataSharing.cs

```diff
@@ -0,0 +1,109 @@
+﻿using ProtoBuf;
+using System;
+using System.IO.Compression;
+using System.IO;
+using Vintagestory.API.Client;
+using Vintagestory.API.Common;
+using ProspectorInfo.Map;
+using System.Linq;
+using System.Text;
+
+namespace ProspectorInfo.Utils
+{
+    internal class ChatDataSharing
+    {
+        private static readonly string CHAT_PREFIX = "ProspectingData;";
+
+        private readonly ICoreClientAPI api;
+        private readonly ProspectorOverlayLayer prospectorOverlayLayer;
+        public ChatDataSharing(ICoreClientAPI api, ProspectorOverlayLayer prospectorOverlayLayer)
+        {
+            this.api = api;
+            this.prospectorOverlayLayer = prospectorOverlayLayer;
+            api.Event.ChatMessage += OnChatMessage;
+        }
+
+        private void OnChatMessage(int groupId, string message, EnumChatType chattype, string data) {
```

Comment:
I wonder if we need to make this more opt-in, as a malicious user could invalidate/destroy the data a user has collected.
I think at least a settings that needs to be activated before the data will be blindly accepted should be added.
Other ideas are time/message limited activation (like you need to press a button to accept the next messages/X minutes)
Or an extra dialog you need to confirm (and maybe optional if your data should only be expanded).

Reply:
Fair point. Maybe one has to explicitly enable an option to receive updates from the chat? Alternatively, we could send the message in a certain group chat, so people have some setup to do before exchanging data.

Reply:
I added an opt-in to accept chat data